core/src/wallet/tezos/synchronizers/TezosLikeAccountSynchronizer.hpp

```diff
@@ -77,17 +77,23 @@ class TezosLikeAccountSynchronizer : public DedicatedContext
     {
         uint64_t blockHeight{0};
         std::string blockHash{""};
-
-        SavedState(uint64_t blockHeight, std::string blockHash)
-            : blockHeight(blockHeight), blockHash(blockHash)
+        uint64_t offset{0};
+
+        SavedState(
+            uint64_t blockHeight,
+            std::string blockHash,
+            uint64_t offset)
+            : blockHeight(blockHeight)
+            , blockHash(blockHash)
+            , offset(offset)
         {}
 
         SavedState() = default;
 
         template<typename Archive>
         void serialize(Archive& archive)
         {
-            archive(blockHeight, blockHash);
+            archive(blockHeight, blockHash, offset);
```

Comment:
I'm not sure about backward compatibility here. Is this working when upgrading or rollbacking libcore versions ?

Reply:
I think the  first synchronization after the upgrade would be a full one (i.e. from block height 0). Anyway, the `SavedState` object is not the same class anymore, so it would probably not be found in the preferences. But after the first synchronization it should be incremental again